backend/src/main/java/fr/openent/appointments/repository/impl/DefaultGridRepository.java

```diff
@@ -145,6 +145,7 @@ public Future<List<Grid>> getGridsWithAvailableTimeSlots(List<Long> gridsIds) {
                 "JOIN " + DB_TIME_SLOT_TABLE + " ts ON ts.grid_id = g.id " +
                 "LEFT JOIN " + DB_APPOINTMENT_TABLE + " a ON a.time_slot_id = ts.id " +
                 "WHERE g.id IN " + Sql.listPrepared(gridsIds) +
+                "AND g.state = '" + GridState.OPEN.getValue() + "' " +

```

Comment:
tu peux juste faire ça non ?
```suggestion
                "AND g.state = ? " +
                ...
                params.add(GridState.OPEN.getValue());
```

Reply:
fait